contracts/modules/royalty/policies/RoyaltyPolicyLAP.sol

```diff
@@ -180,10 +160,51 @@ contract RoyaltyPolicyLAP is
         _initPolicy(ipId, parentIpIds, licenseData);
     }
 
-    /// @notice Returns the Ancestors Vault Implementation address
-    function ancestorsVaultImpl() external view returns (address) {
+    /// @notice Allows the caller to pay royalties to the given IP asset
+    /// @param caller The caller is the address from which funds will transferred from
+    /// @param ipId The ipId of the receiver of the royalties
+    /// @param token The token to pay
+    /// @param amount The amount to pay
+    function onRoyaltyPayment(address caller, address ipId, address token, uint256 amount) external onlyRoyaltyModule {
```

Comment:
Do you need any reentrant protection of this function?  Can you add token more than once?

Reply:
We have an enumerable set that does not allow repeated tokens. When payments are made in a token that already exists in the vault then the token list stays the same - no repetition.

Reply:
nonReentrant may not be needed given the middle external call is to a contract the RoyaltyPolicyLAP deployed so we know the code to not be malicious. In any case we can raise this question to auditors.